File: C  programs/Problems on Pattern Printing/Pattern4.c
/* Write a program accept number from user and Display below pattern.

Input: 5
Output: 2 4 6 8 10

Input: 8
Output: 2 4 6 8 10 12 14 16

*/

#include<stdio.h>

void DisplayPattern(int iNo)
{
   int iCnt=0;
   
   if(iNo<0)
   {
      iNo=-iNo;
   }

   for(iCnt=1;iCnt<=iNo;iCnt++)
   {
      printf("%d\t",iCnt*2);
   }
}

int main()
{
   int iValue=0;

   printf("Enter the number\n");
   scanf("%d",&iValue);

   DisplayPattern(iValue);

   return 0;
}   
   

File: Java programs/Problems on Pattern Printing/Print Star pattern/Pattern4.java
/* Write a java program which Accept the number of rows and number of columns from user and Display below pattern.

Input:  iRow=6  iCol=6
Output : *  *  *  *  *  * 
         *           *  *
         *        *     *
         *     *        *
         *  *           *
         *  *  *  *  *  *                		 

*/
		  
import java.util.*;

class Pattern
{
   void Pattern(int iRow,int iCol)
  {
	   int i=0;
	   int j=0;
	     
		for(i=1;i<=iRow;i++)        
		{
			
			for(j=iCol;j>=1;j--)
		 {
			if ((j==1)||(i==1)||(j==iCol)||(i==iRow))
			{
			 System.out.print("*"+"\t");
			}
			else if(i==j)
			{
			  System.out.print("*"+"\t");
			}
			else
			{
		        System.out.print(" "+"\t");
			}
			
		 }	
		  System.out.println();
        }
   }
}

class Pattern403
{
	
	public static void main(String arg[])
	{
		
		int iValue1=0;
		int iValue2=0;
		
		Scanner sobj=new Scanner(System.in);
		
		System.out.println("Enter number of rows");
		iValue1=sobj.nextInt();
		
		System.out.println("Enter number of columns");
		iValue2=sobj.nextInt();
		
		Pattern pobj=new Pattern();
		pobj.Pattern(iValue1,iValue2);
		
	}	
}